### PR TITLE
[ci] add groups for remaining pipelines

### PR DESCRIPTION
**Pull request by matthewdeng**
## Why are these changes needed?

Follow-up to #36648. Adding groups for remaining pipelines.

**Note:** I used names `pipeline.build.yml` and `pipeline.test.yml` based on the [README](https://github.com/ray-project/ray/blob/master/.buildkite/README.md), but I am not so sure if they are the right description. Feedback welcome.

## Checks

- [ ] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
    - [ ] I've added any new APIs to the API Reference. For example, if I added a 
           method in Tune, I've added it in `doc/source/tune/api/` under the 
           corresponding `.rst` file.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: .buildkite/pipeline.test.yml [hunk 1]
```diff
@@ -1,3 +1,5 @@
+#ci:group=Minimal dependency tests
```
Comment (aslonnie) [inline .buildkite/pipeline.test.yml:1]: I do not know what these tests are, but the tests listed in this file does not say "min deps tests"?

File: .buildkite/pipeline.windows.yml [hunk 2]
```diff
@@ -1,3 +1,5 @@
+#ci:group=Windows tests
```
Comment (aslonnie) [inline .buildkite/pipeline.windows.yml:1]: this yaml file is not (yet) processed by the script that recognizes grouping.

File: .buildkite/pipeline.build.yml [hunk 3]
```diff
@@ -1,3 +1,5 @@
+#ci:group=Rebuild Ray tests
```
Comment (aslonnie) [inline .buildkite/pipeline.build.yml:1]: this is also not accurate.. I can update the readme file in another PR. change it into `pipeline.build*.yml`. 

this is more for tests (or anything) that does not need to run on the container that has ray already built and installed in the environment.

File: .buildkite/pipeline.macos.yml [hunk 4]
```diff
@@ -1,4 +1,4 @@
-#ci:group=MacOS
+#ci:group=MacOS tests
```
Comment (aslonnie) [inline .buildkite/pipeline.macos.yml:1]: this also has builds (wheels and jars), not only tests.

you can add a `:mac: :apple:` prefix if you feel that it is too short and hard to click..

**Review by aslonnie — COMMENTED** [review]
I suggest maybe tolerate this for maybe a month or two.. I am doing the splitting also for migrating tests to the new pipeline, starting with the wheel and docker building parts, as the original `pipeline.build.yml` was too big..

File: .buildkite/pipeline.test.yml (hunk 1 above)
Comment (matthewdeng) [reply .buildkite/pipeline.test.yml:1]: Yeah this was confusing to me as well, but this seems to be the intended use case... 

> pipeline.test.yml contains jobs that only require an installed Ray and a small subset of dependencies, notably exlcuding ML libraries such as Tensorflow or Torch. The tests are run on the test.Dockerfile image.

I don't know the logic for why the minimal dependency tests are in the other pipelines, given this description.

**Pull request closed without merging**